frontend/src/HomePage/HomePage.jsx

```diff
@@ -206,19 +206,21 @@ class HomePage extends React.Component {
                                 <small className="pt-2 app-description">created {moment(app.created_at).fromNow()} ago by {app.user?.first_name} {app.user?.last_name} </small>
                               </td>
                               <td className="text-muted col-auto pt-4">
-                                <Link
+                                {currentUser.role !== 'viewer' && (
```

Comment:
This check should also be present at the router level because if the link is available, the viewer can still access it.
Should enforce the same at backend too. Wdyt?